pkg/storage/watch_cache.go

```diff
@@ -266,7 +266,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
 		return result, nil
 	}
 	if resourceVersion < oldest {
-		return nil, errors.NewInternalError(fmt.Errorf("too old resource version: %d (%d)", resourceVersion, oldest))
+		return nil, errors.NewBadRequest(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest))
```

Comment:
Actually what do you guys think about "412 Precondition Failed"-- I think
410 Gone sounds much too permanent?

On Mon, Nov 2, 2015 at 10:32 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/storage/watch_cache.go
> <https://github.com/kubernetes/kubernetes/pull/16628#discussion_r43662572>
> :
>
> > @@ -266,7 +266,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
> >  		return result, nil
> >  	}
> >  	if resourceVersion < oldest {
> > -		return nil, errors.NewInternalError(fmt.Errorf("too old resource version: %d (%d)", resourceVersion, oldest))
> > +		return nil, errors.NewBadRequest(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest))
>
> Yeah - 410 sounds good.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16628/files#r43662572>.
>